### PR TITLE
Add a password set form to /complete-consents

### PR DESCRIPTION
**Pull request by rahilb**
## What does this change?

Allows a user to set a password when they are registered via an email confirmation link (i.e. we created them a guest account). 

## Screenshots

screen shot 2018-08-13 at 15 15 42

## What is the value of this and can you measure success?

Users without passwords will be given a chance to create one. 

### Does this affect other platforms?

NO

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [x] Locally
- [ ] On CODE (optional)

Related PR: https://github.com/guardian/identity/pull/1340

### CODE REVIEW

**Comment by rahilb** [comment]
Guest password setting metrics are tracked in cloudwatch by IDAPI.

File: identity/app/idapiclient/IdApiClient.scala [hunk 1]
```diff
@@ -158,6 +161,11 @@ class IdApiClient(
   def validateEmail(token: String, trackingParameters: TrackingData): Future[Response[Unit]] =
     post(urlJoin("user","validate-email", token), trackingParameters = Some(trackingParameters)) map extractUnit
 
+  def setPasswordGuest(password: String, token: String): Future[Response[HttpResponse]] = {
```
Comment (QuarpT) [inline identity/app/idapiclient/IdApiClient.scala]: These functions seem trivial, can they be merged?

File: identity/app/idapiclient/IdApiClient.scala [hunk 2]
```diff
@@ -183,6 +191,9 @@ class IdApiClient(
     ) map extract[AccountDeletionResult](identity)
   }
 
+  def put(apiPath: String, auth: Option[Auth] = None, trackingParameters: Option[TrackingData] = None, body: Option[String] = None, extraHeaders: Parameters): Future[Response[HttpResponse]] =
```
Comment (QuarpT) [inline identity/app/idapiclient/IdApiClient.scala]: do we really need this `put` function?

**Review by QuarpT — APPROVED** [review]

**Review by walaura — APPROVED** [review]
plus the flow changes we talked about, LGTM!

**Pull request merged**